### DISCUSSION TITLE
WARNING: The module 'PnP.PowerShell' cannot be installed or updated because it is not a properly-formed module.

### DISCUSSION THREAD

**Discussion by russgove** [pnp/powershell#1813] [General]
Hi, I'm trying to install PnP.Powershell 1.7.10 on a 

Install-Module PnP.PowerShell -RequiredVersion 1.7.0 -Scope CurrentUser  -Force -Verbose -ErrorAction Continue

I get this output
VERBOSE: Using the provider 'PowerShellGet' for searching packages.
VERBOSE: The -Repository parameter was not specified.  PowerShellGet will use all of the registered repositories.
VERBOSE: Getting the provider object for the PackageManagement Provider 'NuGet'.
VERBOSE: The specified Location is 'https://www.powershellgallery.com/api/v2' and PackageManagementProvider is 'NuGet'.
VERBOSE: Searching repository 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='PnP.PowerShell'' for ''.
VERBOSE: Total package yield:'1' for the specified package 'PnP.PowerShell'.
VERBOSE: Performing the operation "Install-Module" on target "Version '1.7.0' of module 'PnP.PowerShell'".
VERBOSE: The installation scope is specified to be 'CurrentUser'.
VERBOSE: The specified module will be installed in 'C:\Users\trwg1\Documents\WindowsPowerShell\Modules'.
VERBOSE: The specified Location is 'NuGet' and PackageManagementProvider is 'NuGet'.
VERBOSE: Downloading module 'PnP.PowerShell' with version '1.7.0' from the repository 'https://www.powershellgallery.com/api/v2'.
VERBOSE: Searching repository 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='PnP.PowerShell'' for ''.
VERBOSE: InstallPackage' - name='PnP.PowerShell', version='1.7.0',destination='C:\Users\trwg1\AppData\Local\Temp\1881477383'
VERBOSE: DownloadPackage' - name='PnP.PowerShell', version='1.7.0',destination='C:\Users\trwg1\AppData\Local\Temp\1881477383\PnP.PowerShell\PnP.PowerShell.nupkg', uri='https://www.powershellgallery.com/api/v2/package/PnP.PowerShell/1.7.0'
VERBOSE: Downloading 'https://www.powershellgallery.com/api/v2/package/PnP.PowerShell/1.7.0'.
VERBOSE: Completed downloading 'https://www.powershellgallery.com/api/v2/package/PnP.PowerShell/1.7.0'.
VERBOSE: Completed downloading 'PnP.PowerShell'.
VERBOSE: Hash for package 'PnP.PowerShell' does not match hash provided from the server.
VERBOSE: InstallPackageLocal' - name='PnP.PowerShell', version='1.7.0',destination='C:\Users\trwg1\AppData\Local\Temp\1881477383'
WARNING: The module 'PnP.PowerShell' cannot be installed or updated because it is not a properly-formed module.

The module does not get installed.

Note that I needed to set my TLS version to 1.2 using 
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
Otherwise I couldnt even connect to the gallery. Not sure if its related.

Any Ideas?

**Comment by gautamdsheth** [comment]
Seems like some issue with PowerShell itself, am transferring this to discussions.

**Reply by russgove** [reply]
I saw from another discussion that it works with PS 7.01. I installed that and its working fine!
